java/client/src/main/java/com/vesoft/nebula/client/graph/storage/general/GeneralStorageClient.java

```diff
@@ -0,0 +1,659 @@
+
+    private GeneralStorageConnection connection;
+    private MetaInfo metaInfo;
+    private MetaClient metaClient;
+    private final List<HostAndPort> addresses;
+    private ExecutorService threadPool;
+    private int timeout = 10000; // ms
+    private final int parallel = 10;
+
+    private final Map<HostAndPort, GeneralStorageConnection> leaderConnections = Maps.newHashMap();
+
+    public GeneralStorageClient(String ip, int port) {
+        this(HostAndPort.fromParts(ip, port));
+    }
+
+    public GeneralStorageClient(HostAndPort address) {
+        this(Arrays.asList(address));
+    }
+
+    public GeneralStorageClient(HostAndPort address, int timeout) {
+        this(Arrays.asList(address), timeout);
+    }
+
+    public GeneralStorageClient(List<HostAndPort> addresses) {
+        this.connection = new GeneralStorageConnection();
+        this.addresses = addresses;
+    }
+
+    public GeneralStorageClient(List<HostAndPort> addresses, int timeout) {
+        this.connection = new GeneralStorageConnection();
+        this.addresses = addresses;
+        this.timeout = timeout;
+    }
+
+    public boolean connect() throws Exception {
+        connection.open(addresses.get(0), timeout);
+        StoragePoolConfig config = new StoragePoolConfig();
+        metaClient = new MetaClient(addresses);
+        metaInfo = metaClient.getMetaInfo();
+        threadPool = Executors.newFixedThreadPool(parallel);
+        return true;
+    }
+
+    /**
+     * Put key-value into nebula
+     *
+     * @param spaceName nebula graph space
+     * @param key       nebula key
+     * @param value     nebula value
+     * @return boolean
+     */
+    public boolean put(String spaceName, String key, String value) throws Exception {
+        int spaceId = metaInfo.getSpaceNameMap().get(spaceName);
+        int part = keyToPartId(spaceName, key);
+
+        Map<Integer, List<KeyValue>> parts = Maps.newHashMap();
+        List<KeyValue> pairs = Lists.newArrayList(new KeyValue(key.getBytes(), value.getBytes()));
+        parts.put(part, pairs);
+        KVPutRequest request = new KVPutRequest(spaceId, parts);
+
+        HostAndPort leader = getLeader(spaceName, part);
+        return doPut(spaceName, leader, request);
+    }
+
+
+    /**
+     * Put multi key-value into nebula
+     *
+     * @param spaceName nebula graph space
+     * @param kvs       key-value pairs
+     * @return boolean
+     */
+    public boolean put(String spaceName, Map<String, String> kvs) {
+        int spaceId = metaInfo.getSpaceNameMap().get(spaceName);
+        Map<Integer, List<KeyValue>> partKeyValues = Maps.newHashMap();
+        for (Map.Entry<String, String> kv : kvs.entrySet()) {
+            int part = keyToPartId(spaceName, kv.getKey());
+            KeyValue keyValue = new KeyValue(kv.getKey().getBytes(), kv.getValue().getBytes());
+            if (!partKeyValues.containsKey(part)) {
+                partKeyValues.put(part, new ArrayList<KeyValue>());
```

Comment:
add `continue;`

Reply:
why continue? if use continue here, then one part only has one KeyValue.

Reply:
I misread that when list was initialized, has put data in it.